rustls/tests/api.rs

```diff
@@ -7155,6 +7155,77 @@ fn test_refresh_traffic_keys() {
     check_both_directions(&mut client, &mut server);
 }
 
+#[test]
+fn test_automatic_refresh_traffic_keys() {
+    static LIMITED_SUITE: OnceLock<rustls::Tls13CipherSuite> = OnceLock::new();
+    let limited = LIMITED_SUITE.get_or_init(|| {
+        with_confidentiality_limit(provider::cipher_suite::TLS13_AES_128_GCM_SHA256, 1024)
```

Comment:
Is there a way to write this that would be a little bit less brittle w.r.t some of the constants?

An easy change would be to lift the 1024 limit from the `limited` ciphersuite into a binding that could be ref'd for the `for` range on L7186 and the assert on L7204.

It might be nice if the expected `transfer` values (30 and 57) could be calculated based on the actual encoding of some exemplar messages but that feels pretty heavy-weight. I would be OK with how it is now.

Reply:
So I have hoisted a lot of this into `with_confidentiality_limit` (and renamed it). The remainder has `const CONFIDENTIALITY_LIMIT: usize = 1024;` plus a const fn that gives how big a TLS message is with a given payload.

Reply:
Looks much nicer, thank you!